git.zsh

```diff
@@ -35,11 +37,85 @@ _fzf_complete_git() {
     fi
 
     if [[ "$@" = 'git commit'* ]]; then
-        if [[ "$prefix" = '--fixup=' ]]; then
+        if [[ "$prefix" =~ '^--(reuse-message|reedit-message|fixup|squash)=' ]]; then
+            prefix_option="${prefix/=*/=}"
+            _fzf_complete_git-commits '' "$@"
+            unset prefix_option
+            return
+        fi
+
+        if [[ "$last_options" =~ '^(-[^-]*[cC]|--(reuse-message|reedit-message|fixup|squash))$' ]]; then
             _fzf_complete_git-commits '' "$@"
+            return
         else
+        fi
+
+        if [[ "$prefix" =~ '^--message=' ]]; then
+            prefix_option="${prefix/=*/=}"
             _fzf_complete_git-commit-messages '' "$@"
+            unset prefix_option
+            return
+        fi
+
+        if [[ "$last_options" =~ '^(-[^-]*m|--message)$' ]]; then
+            _fzf_complete_git-commit-messages '' "$@"
+            return
+        fi
+
+        if [[ "$prefix" =~ '^--author=' ]]; then
+            return
+        fi
+
+        if [[ "$prefix" =~ '^--date=' ]]; then
+            return
+        fi
+
```

Comment:
`--author 'Foo Bar <foobar@example.com>'` or `--date 'Mon, 13 Jan 2020 20:42:53 +0900'` should have no completion as well.